sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1293,7 +1293,7 @@ object SQLConf {
       .createWithDefault(true)
 
   val STREAMING_METRICS_ENABLED =
-    buildConf("spark.sql.streaming.metricsEnabled")
+    buildConf("spark.metrics.sparkStreamingSource.enabled")
```

Comment:
It's a breaking change to rename an already released config. Can we leave it as it is?

Reply:
Or, what about renaming together at 3.0? We still can support legacy config names together, cannot we?
Also, cc @gatorsmile since he suggest renaming.

Reply:
The old config would have to be deprecated and be an alias for the new one, at least.

Reply:
Yep. +1 for that.

Reply:
I fully agree on the need to deprecate the old parameter and adding an alias.
Another point I see about renaming `spark.sql.streaming.metricsEnabled` to `spark.metrics.sparkStreamingSource.enabled`, is that the configuration lives in [[SQLConf]] where all configuration parameters appear to have `spark.sql` as their prefix. For example, would using `spark.sql.metrics.sparkStreamingSource.enabled` be a better choice? I see pros and cons to that.  Any advice?

Reply:
Adding the voice of name name here: we may want to differentiate DStream and Structured Streaming. Before the change it is clear that it's for Structured Streaming (sql.streaming) which new name is not clear about that. If we want to only mention `streaming` it should also mention `sql`. If we are OK to have `structuredStreaming` as a part of name, that's fine.